go/services/reports/internal/services/reports_service.go

```diff
@@ -43,14 +45,18 @@ type ReportsServerParams struct {
 }
 
 func NewReportsService(p ReportsServerParams) *ReportsService {
-	return &ReportsService{
+	reportsService := &ReportsService{
 		logger:                       p.Logger,
 		reportRepository:             p.ReportRepository,
 		applicationLogsRepository:    p.ApplicationLogsRepository,
 		nodeLogsRepository:           p.NodeLogsRepository,
 		applicationInsightsGenerator: p.ApplicationInsightsGenerator,
 		nodeInsightsGenerator:        p.NodeInsightsGenerator,
 	}
+
+	go reportsService.PollReports(context.Background())
```

Comment:
Starting polling in the constructor is quite unexpected, maybe it can be extracted to some wrapper function like PollForReports()?

Reply:
Yeah, that's a good point, I will move it to the root constructor